src/coreclr/jit/fgopt.cpp

```diff
@@ -493,70 +279,21 @@ bool Compiler::fgRemoveUnreachableBlocks(CanRemoveBlockBody canRemoveBlock)
 //
 //   Assumes the predecessor lists are computed and correct.
 //
-//   Use `fgReachable()` to check reachability.
-//   Use `fgDominate()` to check dominance.
-//
 PhaseStatus Compiler::fgComputeReachability()
 {
     assert(fgPredsComputed);
 
-    fgComputeReturnBlocks();
-
-    // Compute reachability and then delete blocks determined to be unreachable. If we delete blocks, we
-    // need to loop, as that might have caused more blocks to become unreachable. This can happen in the
-    // case where a call to a finally is unreachable and deleted (maybe the call to the finally is
-    // preceded by a throw or an infinite loop), making the blocks following the finally unreachable.
-    // However, all EH entry blocks are considered global entry blocks, causing the blocks following the
-    // call to the finally to stay rooted, until a second round of reachability is done.
-    // The dominator algorithm expects that all blocks can be reached from the fgEnterBlks set.
-    unsigned passNum = 1;
-    bool     changed;
-
-    auto canRemoveBlock = [&](BasicBlock* block) -> bool {
-        // If any of the entry blocks can reach this block, then we skip it.
-        if (!BlockSetOps::IsEmptyIntersection(this, fgEnterBlks, block->bbReach))
-        {
-            return false;
-        }
-
-        return true;
-    };
-
-    bool madeChanges = false;
+    bool madeChanges = fgDfsBlocksAndRemove() != PhaseStatus::MODIFIED_NOTHING;
 
-    do
-    {
-        // Just to be paranoid, avoid infinite loops; fall back to minopts.
-        if (passNum > 10)
-        {
-            noway_assert(!"Too many unreachable block removal loops");
-        }
-
-        // Walk the flow graph, reassign block numbers to keep them in ascending order.
-        JITDUMP("\nRenumbering the basic blocks for fgComputeReachability pass #%u\n", passNum);
-        passNum++;
-        madeChanges |= fgRenumberBlocks();
-
-        //
-        // Compute fgEnterBlks, reverse post-order, and bbReach.
-        //
+    madeChanges |= fgRenumberBlocks();
 
-        fgComputeEnterBlocksSet();
-        fgDfsReversePostorder();
-        fgComputeReachabilitySets();
-
-        //
-        // Use reachability information to delete unreachable blocks.
-        //
-
-        changed = fgRemoveUnreachableBlocks(canRemoveBlock);
-        madeChanges |= changed;
+    fgDfsReversePostorder();
 
-    } while (changed);
+    // fgDfsReversePostorder reassigns preorder numbers, so recompute the DFS.
+    m_dfsTree = fgComputeDfs();
```

Comment:
We gain a new DFS tree computation here, but it should be quite short lived; we'll be able to get rid of it once the old DFS is removed.